x-pack/plugins/apm/public/components/app/transaction_details/transaction_details_tabs.tsx

```diff
@@ -46,12 +48,18 @@ const tabs = [
 export function TransactionDetailsTabs() {
   const { query } = useApmParams('/services/{serviceName}/transactions/view');
 
+  const isCriticalPathFeatureEnabled = useCriticalPathFeatureEnabledSetting();
+
+  const availableTabs = isCriticalPathFeatureEnabled
+    ? tabs.concat(aggregatedCriticalPathTab)
+    : tabs;
```

Comment:
Can't you use just the tabs variable?
```suggestion
  if(isCriticalPathFeatureEnabled){
    tabs.push(aggregatedCriticalPathTab)
  }
```

Reply:
That would add a tab on every render. I think this is fine.

Reply:
Agree, it's fine as is. If we get more conditional tabs, something like this makes more sense:

```suggestion
  const availableTabs = [
    ...tabs,
    ...(isCriticalPathFeatureEnabled ? [aggregatedCriticalPathTab] : []),
  ];
```

But for now there's no difference.